Controller/Order/Save.php

```diff
@@ -103,13 +103,20 @@ public function __construct(
     public function execute()
     {
         try {
+            $sessionQuote = $this->checkoutSession->getQuote();
             // get the transaction reference parameter
             $reference = $this->getRequest()->getParam('reference');
             // call order save and update
             list($quote, $order) = $this->orderHelper->saveUpdateOrder($reference);
```

Comment:
@danac-gs Could you please help me when you have a chance?
Is it any chance that saveUpdateOrder method can return empty $quote?

I am asking because I see this check. We check if parentQuote is empty and few lines below we return it as $quote.
https://github.com/BoltApp/bolt-magento2/blob/7cb86c96ad800cc60034d48f8136a8655eea7e92/Helper/Order.php#L791

Reply:
@vitaliyreznikov at some point all of the inactive quotes (our parent quote in this case) get deleted via cron job, by default after a month I think. So yes, an empty quote ("false" actually) can be returned from that call but not in your use case, calling order save right after the purchase.

Reply:
Thank you, Daniel. So, what do you think about this PR? Just need fix/add tests?